app/src/main/java/io/appium/uiautomator2/handler/GetBatteryInfo.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.uiautomator2.handler;
+
+import android.app.Instrumentation;
+import android.support.test.InstrumentationRegistry;
+
+import org.json.JSONObject;
+
+import io.appium.uiautomator2.handler.request.SafeRequestHandler;
+import io.appium.uiautomator2.http.AppiumResponse;
+import io.appium.uiautomator2.http.IHttpRequest;
+import io.appium.uiautomator2.server.WDStatus;
+import io.appium.uiautomator2.utils.BatteryHelper;
+import io.appium.uiautomator2.utils.Logger;
+
+public class GetBatteryInfo extends SafeRequestHandler {
+    private final Instrumentation mInstrumentation = InstrumentationRegistry.getInstrumentation();
+
+    public GetBatteryInfo(String mappedUri) {
+        super(mappedUri);
+    }
+
+    @Override
+    public AppiumResponse safeHandle(IHttpRequest request) {
+        Logger.info("Get Battery Info command");
+        final JSONObject response = new JSONObject();
+        final BatteryHelper batteryHelper = new BatteryHelper(mInstrumentation
+                .getTargetContext());
+        try {
+            response.put("level", batteryHelper.getLevel());
+            response.put("status", batteryHelper.getStatus());
```

Comment:
Can we change the response name of `status` to `state` like iOS? 
Since when I implemented it in Ruby client, I faced https://github.com/appium/ruby_lib_core/pull/85#discussion_r185284872 error and it failed to parse `status` as an error code of JsonWiredProtocol.

Reply:
I just wanted to keep system-specific terminology. In Android this indicator is called STATUS

Reply:
if we can, it'd be nice to have cross-platform compatibility for the API. is that possible here?

Reply:
ok, I'll change rename than on driver level